### PR TITLE
DPR-618: Review logs and minor refactoring

### PR DESCRIPTION
**Pull request by koladeadewuyi-moj**
This PR reviews the logs and also some minor refactoring.

### CODE REVIEW

**Review by dominic-messenger-moj — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/digital/writer/Writer.java [hunk 1]
```diff
@@ -74,16 +74,18 @@ protected static Consumer<Row> processRow(
...
+            String unvalidatedOperation = row.getAs(OPERATION);
+            val optionalOperation = getOperation(unvalidatedOperation);
+
             if (optionalOperation.isPresent()) {
                 val operation = optionalOperation.get();
                 try {
                     writeRow(spark, storage, destinationPath, primaryKey, operation, row);
                 } catch (DataStorageException ex) {
-                    logger.warn("Failed to {}: {} to {}", operation.getName(), row.json(), destinationPath);
+                    logger.error("Failed to {}", String.format("%s: %s to %s", operation.getName(), row.json(), destinationPath), ex);
```
Comment (carwynmoj) [inline src/main/java/uk/gov/justice/digital/writer/Writer.java:85]: Hmm...the `{}` interpolation should work here without you having to use a `String.format` like line 88 below.

File: src/main/java/uk/gov/justice/digital/service/DataStorageService.java [hunk 2]
```diff
@@ -146,45 +149,43 @@ public void resync(String tablePath, Dataset<Row> df) throws DataStorageExceptio
...
     }
 
     public Dataset<Row> get(SparkSession spark, String tablePath) {
-        DeltaTable deltaTable = getTable(spark, tablePath);
-        return deltaTable == null ? null : deltaTable.toDF();
+        return getTable(spark, tablePath).map(DeltaTable::toDF).orElse(spark.emptyDataFrame());
     }
 
-    protected DeltaTable getTable(SparkSession spark, String tablePath) {
+    protected Optional<DeltaTable> getTable(SparkSession spark, String tablePath) {
```
Comment (carwynmoj) [inline src/main/java/uk/gov/justice/digital/service/DataStorageService.java:183]: 👍

**Review by carwynmoj — APPROVED** [review]
LGTM 👍 

One minor comment on one of the log lines. You shouldn't have to use a `String.format` I don't think.

**Pull request merged**